modin/pandas/test/test_series.py

```diff
@@ -509,6 +509,8 @@ def test___repr__(name, dt_index, data):
     else:
         assert repr(modin_series) == repr(pandas_series)
 
+    df_equals(modin_series.index, pandas_series.index)
```

Comment:
Why do we add this check? Original plan was in adding check on equality of `df.index` and `df._query_compiler.to_pandas().index`, described here https://github.com/modin-project/modin/pull/3141#issuecomment-872258328

Reply:
@Rubtsowa , this check should go right after 508 line as I specified [here](https://github.com/modin-project/modin/pull/3212#discussion_r663693092) because to create an empty DF `to_pandas` is called for BaseOnPython only.
@prutskov , this check covers that case.

Reply:
This check doesn't cover that case for the other backends, because original issue was in missing correct index of pandas DataFrame after `PandasFrame.to_pandas()` method. The other backends hided `to_pandas` issue, because `df.index` don't provoke call of `to_pandas` underneath.

Reply:
The issue was related to `to_pandas`. We need to verify that Modin Dataframe index after `to_pandas` call is equal to pandas Dataframe index. The call `modin_df._query_compiler.to_pandas().index` for each backend is the same if we call `df.index` for `BaseOnPython` since `to_pandas` is called underneath.

Reply:
I agree with you, that "The call `modin_df._query_compiler.to_pandas().index` for each backend is the same if we call `df.index` for `BaseOnPython` for empty frames since the same `to_pandas` is called underneath." My suggestion is mainly directed at highlighting this issue for other backends.

Reply:
Okay, let's add the test. @Rubtsowa, can you do it, please?

Reply:
Also, fix your commit message (remove spaces in "TEST - #3214").